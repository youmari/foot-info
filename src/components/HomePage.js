import { CogIcon, MicrophoneIcon } from '@heroicons/react/solid';
import { GlobeIcon } from '@heroicons/react/outline';
import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { v4 as uuid } from 'uuid';
import fetchData from '../api/api';
import { fetchCountriesStatus, getCountries, getCountriesFailure } from '../redux/homePage/homePage';
import Country from './Country';

const HomePage = () => {
  const [searchInput, setSearchInput] = useState('');
  const dispatch = useDispatch();
  const countriesData = useSelector((state) => state.countries.onSuccess);
  const filtredCountriesData = countriesData?.filter((country) => (
    country.name.toLowerCase().includes(searchInput.toLocaleLowerCase())
  ));
  useEffect(async () => {
    if (!countriesData.length) {
      try {
        dispatch(fetchCountriesStatus({ bool: true }));
        const { data: { response } } = await fetchData('countries');
        dispatch(fetchCountriesStatus({ bool: false }));
        dispatch(getCountries(response));
      } catch (err) {
        dispatch(getCountriesFailure({ err }));
      }
    }
  }, []);

  return (
    <div className="bg-blue-700">
      <div className="flex justify-end bg-blue-800">
        <MicrophoneIcon className="w-5 text-white cursor-pointer" />
        <CogIcon className="w-5 text-white m-3 ml-6 mr-3 cursor-pointer" />
      </div>
      <div className="flex justify-center items-center p-4">
        <GlobeIcon className="w-14 text-white" />
        <h2 className="text-2xl font-gill font-bold text-white">World Leagues info</h2>
      </div>
      <div className="flex justify-center bg-blue-700 m-3">
        <input className="w-13 rounded font-gill text-base p-2 border-solid border-blue-900 text-blue-900 placeholder:text-blue-900" type="search" placeholder="Search by Country..." onChange={(e) => setSearchInput(e.target.value)} />
      </div>
      <h2 className="font-gill bg-blue-800 text-white text-sm  pl-2">Leagues by Countries</h2>
      <div className="grid grid-cols-2 h-full bg-blue-700">
        {filtredCountriesData?.map((country) => (
          <Country key={uuid()} country={country} />
        ))}
      </div>
    </div>
  );
};

export default HomePage;
