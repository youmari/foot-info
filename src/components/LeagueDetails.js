import { useSelector } from 'react-redux';
import { Route, Routes } from 'react-router-dom';
import { v4 as uuid } from 'uuid';
import DetailsPage from './DetailsPage';

const LeagueDetails = () => {
  const countriesData = useSelector((state) => state.countries.onSuccess);

  return (
    <>
      {countriesData?.map((country) => (
        <Routes key={uuid()}>
          <Route path={country.name} element={<DetailsPage country={country} />} />
        </Routes>
      ))}
    </>
  );
};

export default LeagueDetails;
