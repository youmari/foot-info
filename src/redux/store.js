import { applyMiddleware, createStore } from 'redux';
import logger from 'redux-logger';
import footBallReducer from './homePage/homePage';

// const reducers = combineReducers(
//   footBallReducer,
// );
const middlewares = [logger];

const store = createStore(footBallReducer,
  applyMiddleware(...middlewares));

export default store;
