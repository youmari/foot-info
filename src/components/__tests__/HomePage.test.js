import {
  fireEvent,
  render, screen, waitFor,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import store from '../../redux/store';
import HomePage from '../HomePage';
import LeagueDetails from '../LeagueDetails';

describe('HomePage test', () => {
  test('should HomePage match the snapchot', async () => {
    const app = render(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route element={<HomePage />} path="/" />
          </Routes>
        </Router>
      </Provider>,
    );
    expect(app).toMatchSnapshot();
  });

  test('should render and and find Brazil link', async () => {
    render(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route element={<HomePage />} path="/" />
          </Routes>
        </Router>
      </Provider>,
    );
    const brazilLeague = await waitFor(() => screen.getByRole('link', { name: /country flag brazil br/i }));
    expect(brazilLeague).toBeInTheDocument();
  });

  test('should render and and find Brazil link and click it and get the first league ', async () => {
    render(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route element={<HomePage />} path="/" />
          </Routes>
          <LeagueDetails />
        </Router>
      </Provider>,
    );
    const canadaLeague = await waitFor(() => screen.getByRole('link', { name: /country flag canada ca/i }));
    fireEvent.click(canadaLeague);
    const league = await waitFor(() => screen.getByRole('heading', { name: /canadian championship/i }));
    expect(league).toBeInTheDocument();
  });
});
