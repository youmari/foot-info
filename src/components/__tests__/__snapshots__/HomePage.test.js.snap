// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage test should HomePage match the snapchot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="font-gill bg-blue-800 text-white text-sm  pl-2"
      >
        Leagues by Countries
      </h2>
      <div
        class="grid grid-cols-2 h-full bg-blue-700"
      />
    </div>
  </body>,
  "container": <div>
    <h2
      class="font-gill bg-blue-800 text-white text-sm  pl-2"
    >
      Leagues by Countries
    </h2>
    <div
      class="grid grid-cols-2 h-full bg-blue-700"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
